### PR TITLE
Add out_dtype support for sparse semi-structured CUTLASS back-end

### PR DESCRIPTION
**Pull request by alexsamardzic**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116519

cc @nikitaved @pearu @cpuhrsch @amjames @bhosmer @jcaip @ptrblck

### CODE REVIEW

**Comment by alexsamardzic** [comment]
This PR is adding `out_dtype` support for CUTLASS back-end for sparse semi-structured GEMM, to match corresponding functionality provided by cuSPARSELt back-end.  It is a step towards fixing #115420.

File: test/test_sparse_semi_structured.py [hunk 1]
```diff
@@ -335,7 +333,7 @@ def test_mm_sparse_second_NT(self, dense_input_shape, dtype, device, backend):
 
         # Currently we don't support int matmul on GPU, so evaluate on CPU and copy over
```
Comment (vadimkantorov) [inline test/test_sparse_semi_structured.py:334]: maybe now there is torch._int_mm on GPU as well: https://github.com/pytorch/pytorch/issues/107671?

Comment (alexsamardzic) [reply test/test_sparse_semi_structured.py:334]: PR is not touching this part of the test; not sure about `_int_mm()`, will defer it for @jcaip.

**Review by cpuhrsch — APPROVED** [review]

**Pull request closed without merging**